src/kontrol/solc_to_k.py

```diff
@@ -64,6 +64,84 @@ def solc_to_k(
     return _kprint.pretty_print(bin_runtime_definition, unalias=False) + '\n'
 
 
+@dataclass(frozen=True)
+class Input:
+    parent_name: str
+    name: str
+    type: str
+    components: list[Input] = field(default_factory=list)
+
+    @staticmethod
+    def from_dict(_input: dict) -> Input:
+        name = _input['name']
+        type = _input['type']
+        if _input.get('components') is not None:
+            if type == 'tuple':
+                components = Input._recurse_comp(name, _input['components'])
+            else:
+                dimension = _find_array_dimensions(type)
+                components = Input._recurse_comp(name + "[]" * dimension, _input['components'])
+            return Input("", name, type, components)
+        else:
+            return Input("", name, type)
+
+    @staticmethod
+    def _recurse_comp(parent: str, components: dict) -> list[Input]:
+        comps = []
+        for comp in components:
+            _name = comp['name']
+            _type = comp['type']
+            if comp.get('components') is not None:
+                if _type == 'tuple':
+                    new_comps = Input._recurse_comp(parent, _name, comp['components'])
```

Comment:
Input.`_recurse_comp` is defined above with two arguments, yet here the call has three.